frontends/concrete-python/examples/xor_distance/hamming_distance.py

```diff
@@ -0,0 +1,217 @@
+import argparse
+import time
+
+import numpy
+
+from concrete import fhe
+
+
+# Hamming weight computation
+def hw(x):
+
+    # Hamming Weight table for 8b entries
+    # fmt: off
+    hw_table_ref = [
+        0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3,
+        4, 3, 4, 4, 5, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4,
+        4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2,
+        3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5,
+        4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 1, 2, 2, 3, 2, 3, 3,
+        4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3,
+        3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5,
+        6, 6, 7, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6,
+        4, 5, 5, 6, 5, 6, 6, 7, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 4, 5, 5,
+        6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8]
+    # fmt: on
+
+    hw_table = fhe.LookupTable(hw_table_ref)
+    return hw_table[x]
+
+
+# Reference function for tests
+def dist_in_clear(x, y):
+
+    return numpy.sum(hw(x ^ y))
+
+
+# For all bitsize_w
+def dist_directly_from_cp(x, y):
```

Comment:
yes, using bit extraction by chunks of 2 bits, do the bit count on xor on 2x2bits with  4b TLU, sum all the results.

Reply:
I think you can extract extra constant chunks for free, to support all precisions. Not sure how you handle the TLU on constants. @umut-sahin  ?

Reply:
wouldn't it be what we have in `dist_with_xor_internal(x, y, bitsize_w=2)` or worse (dist_with_xor_internal assumes the chunks are done pre-encryption)?

Reply:
(changed a bit the function names for coherence)

Reply:
@rudy-6-4 : I think your idea is already included in dist_with_xor_internal for bitsize_w=2, do you agree?

Reply:
I don't think so. I see no bit extraction (`fhe.bits`) and for bitwize_w = 4 the tlu is a 8bit one instead of 2 4bits ones.
I would expect something like (untensorized):

```python
full_size = 16
chunk_size = 2
chunks_index = range(0, full_size, chunk_size)
chunks_x = [ fhe.bits(x)[i:i+chunk_size] for i in chunks_index] ]
chunks_y = [ fhe.bits(y)[i:i+chunk_size] for i in chunks_index] ]
chunks_xor_x_y = [ xor_chunk_tlu(chunk_x, chunk_y) for chunk_x, chunk_y in zip(...) ]
return sum(chunks_xor_x_y)
```
where xor_chunk_tlu is using fhe.multivariate (or the packing +TLU as you do)

Reply:
I don't see how what you say would differ from using `dist_with_xor_internal` with `bitsize_w = 4`. Let's talk about it in a meeting?

Indeed, in my config:
- no need to calls fhe.bits, things are directly by groups of 4b
- automatically, `table = fhe.LookupTable([hw((i % power) ^ (i // power)) for i in range(power**2)])` would make TLU's of 4b

But you're right that I could use `fhe.multivariate` instead of doing ` = x + power * y` myself. I would say that internally in CP, it should behave the same way, however.